pkg/runtime/reconciler.go

```diff
@@ -638,13 +668,15 @@ func (r *resourceReconciler) patchResourceStatus(
 	}()
 
 	rlog.Enter("kc.Patch (status)")
-	err = r.kc.Status().Patch(
-		ctx,
-		latest.RuntimeObject(),
-		client.MergeFrom(desired.DeepCopy().RuntimeObject()),
-	)
+	dobj := desired.DeepCopy().RuntimeObject()
+	lobj := latest.DeepCopy().RuntimeObject()
+	patch := client.MergeFrom(dobj)
```

Comment:
It may be fine to do this, because I don't believe patching status updates the `resourceVersion`